code/hsec-tools/src/Security/Advisories/Generate/HTML.hs

```diff
@@ -257,11 +258,25 @@ feed advisories =
         <> Advisories.advisorySummary advisory
     toUrl advisory = advisoriesRootUrl <> "/" <> advisoryLink (Advisories.advisoryId advisory)
 
-renderFeed :: [Advisories.Advisory] -> Text
-renderFeed =
+renderFeed :: UTCTime -> [Advisories.Advisory] -> Text
+renderFeed now =
   maybe (error "Cannot render atom feed") TL.toStrict
     . FeedExport.textFeed
     . feed
+    . sortAndDropOld now
+
+-- | Sort advisories by modification date (recent updates first).
+-- Then, keep all advisories that were modified in the last month.
+-- Finally, if there are fewer than 10 such advisories, top up the
+-- list to include 10 entries.
+--
+sortAndDropOld :: UTCTime -> [Advisories.Advisory] -> [Advisories.Advisory]
+sortAndDropOld now =
+  (\(h,t) -> h <> take (10 - length h) t)
+  . span ((> thirtyDaysAgo) . zonedTimeToUTC . Advisories.advisoryModified)
+  . sortOn (Down . zonedTimeToUTC . Advisories.advisoryModified)
+  where
+    thirtyDaysAgo = addUTCTime (negate (nominalDay * 30)) now
```

Comment:
I feel that it's quite confusing/redundant.

I use a feed reader and whenever I add a new feed, I have the last 10 articles marked as "unseen", it gives an explicit health status of the feed (i.e. "old entries = not really active" vs "no entries = broken ?")

Reply:
I think so too. In fact, I get upset when I add a feed and only the most recent posts are there and then I have to go back to the source and find out the other posts.

I think we should wait and see if the feed becomes too long that this becomes an issue. We can also alleviate this problem (at that time) by reducing the amount of text included in the feed, instead of the full content we can only include a short summary / start of entry?

Reply:
Fair enough; thanks for the feedback.  Consensus is not to land it so I shall close the ticket.  If it becomes an issue later we could offer a "recent" feed in addition to complete feed.